### PR TITLE
update auto re login

### PR DESCRIPTION
**Pull request by arunstop**
- update : authing now uses cookie instead of local storage
- update : user cookies now uses firebase auth data instead of the obsolete firestore data
- update : added dotenv and firebase-admin package
- updated : using firebase auth data instead of obsolete firestore data when loggin in via cookie
- update : added user add module
- fix : firebase auth data `User` now only shows in local machine ,,, also handling an error where auth data exists but not the firestore one on register
- update : added session record when login to realtimedatabase
- update : added dateUpdated on users in firestore
- update : added synchronization of dateUpdated on user's firestore and user's session on realtime db
- update : added auth validation via server-side with route api
- update : added re-login feature

### CODE REVIEW

**Comment by arunstop** [comment]
Attempted to validate in server side but couldn't get it done, i guess we can leave it as is, not that big of a deal anyway

**Comment by scuffed-developer** [comment]
Okay, i guess the commit [https://github.com/arunstop/scuffed-blogger/pull/11/commits/2578f5d4c8afebd153c9ad23cbcecb713bf6fdf2](update : added auth validation via server-side with route api) is impactless, but i guess we can keep it there.

**Comment by arunstop** [comment]
Maybe we'll start a new issue to login/re-login via server-side.

Here are some pseudo-code : 
=> Login
Firebase auth -> Login
Firestore -> get data
Rtdb -> add session
combine all 3 data
save to cookies

=> Re-login
Get cookies
Firebase auth -> Validate
IF Validate == TRUE
	THEN check latest session
	IF latest session != cookies latest session
	THEN get data from firestore
	THEN re-login again
	THEN save data to cookies
	END
IF Validate == FALSE
	THEN DO => Login

File: utils/contexts/auth/AuthProvider.tsx [hunk 1]
```diff
@@ -63,12 +75,33 @@ export const AuthProvider = ({
...
+
+        console.log((localUserData?.localAuthData as any).lastLoginAt);
+
+        if (
+          (user.metadata as any).lastLoginAt !==
+          (localUserData?.localAuthData as any).lastLoginAt
+        ) {
+          // getting new data
+          // console.log("getting new data");
+          const newUserData = await fbUserGet({ email: localUserData.email });
```
Comment (scuffed-developer) [inline utils/contexts/auth/AuthProvider.tsx:93]: Noted for future re-login fix using server-side

**Comment by scuffed-developer** [comment]
That's good, but i think  we need to kinda do the other feature first.

Just to make sure, validating authentication is just for the re-login right? 

So then i made an issue [server side authentication](https://github.com/arunstop/scuffed-blogger/pull/11#discussion_r935853034_)

**Comment by arunstop** [comment]
OK, nice please do assing me as well!

File: utils/contexts/auth/AuthProvider.tsx (hunk 1 above)
Comment (scuffed-developer) [reply utils/contexts/auth/AuthProvider.tsx:93]: Just for formality please do answer this so i can proceed @arunstop

Comment (arunstop) [reply utils/contexts/auth/AuthProvider.tsx:93]: Okay noted already thanks my dear variant!

**Review by scuffed-developer — APPROVED** [review]
Seems good proceeding now, please do merge!

**Pull request merged**